test/python/quantum_info/states/test_statevector.py

```diff
@@ -41,6 +41,61 @@
 class TestStatevector(QiskitTestCase):
     """Tests for Statevector class."""
 
+    def test_from_circuit_bell_state(self):
+        """Test from_circuit creates Bell state correctly."""
+        qc = QuantumCircuit(2)
+        qc.h(0)
+        qc.cx(0, 1)
+
+        sv = Statevector.from_circuit(qc)
+        expected_data = np.array([1 / np.sqrt(2), 0, 0, 1 / np.sqrt(2)])
+        expected = Statevector(expected_data)
+
+        self.assertTrue(sv.equiv(expected))
+
+    def test_from_circuit_transpilation_consistency(self):
+        """Test transpilation consistency - the main GitHub issue."""
```

Comment:
Since this is not a bugfix we don't have to link the issue here. If people are interested they can find the original issue via git blame to get to this PR and then to the issue 🙂